server/core_link.go

```diff
@@ -44,15 +45,15 @@ func LinkApple(ctx context.Context, logger *zap.Logger, db *sql.DB, config Confi
 	}
 
 	res, err := db.ExecContext(ctx, `
-UPDATE users
-SET apple_id = $2, update_time = now()
+UPDATE users AS u
+SET apple_id = $2, email = COALESCE(u.email, $3), update_time = now()
```

Comment:
I would suggest we adjust part of that query to be: `COALESCE(NULLIF(u.email, ''), $3)`. This way we can set the `$3` as the email address even if the existing email field is empty rather than NULL.